### PR TITLE
jmapgen itemgroup item default chance

### LINKED ISSUE #72326: Palletized(/nested?) item placement without defined "chance" defaults to 0 (will never be placed)
### Describe the bug

We have a palette (`lodge_items`)with a definition `"L": { "item": "cannibal_food", "repeat": [ 1, 4 ] },`.

https://github.com/CleverRaven/Cataclysm-DDA/blob/ef537eb9982dd516ab1d78923475f8913d5660ab/data/json/mapgen_palettes/lodge_palette.json#L159-L162

Natural understanding from repeating that is that it will spawn 1-4 of those items. Instead, it always spawns 0.

Note that `"D": { "item": "hunting_lodge_weapons" },` and `"h": { "item": "cannibal_weapons", "repeat": [ 1, 4 ] },` will never be placed either, lacking a defined `chance`

Here is an image of a nested mapgen spawn from the debug menu:

image

And for comparison, the mapgen entry: https://github.com/CleverRaven/Cataclysm-DDA/blob/ef537eb9982dd516ab1d78923475f8913d5660ab/data/json/mapgen/nested/lodge_nested.json#L20-L22

Notice that every square with L and D defined is empty. They are always empty. They will never spawn, because their chance defaults to 0.

### Attach save file

N/A requires modification to prove without a doubt. Use this branch: https://github.com/CleverRaven/Cataclysm-DDA/compare/master...RenechCDDA:Cataclysm-DDA:nested_mapgen_fail

Debug spawn lodge, teleport to lodge. Notice that the pantry is empty despite having many definitions for items to be put into it. Notice that the only change to files has been to remove explicit `chance` definitions.

### Steps to reproduce

Visit cannibal lodge, notice that there is nothing from the `cannibal_food` entry despite it being defined.

Use this branch to verify: https://github.com/CleverRaven/Cataclysm-DDA/compare/master...RenechCDDA:Cataclysm-DDA:nested_mapgen_fail

### Expected behavior

If I define an item to spawn in mapgen without specifying a chance, it should default to 100, definitely not 0! I defined it to spawn, didn't I?

### Screenshots

_No response_

### Versions and configuration

c5225cf13d0114c4c4429b765e19093b83e5e5a3

Windows 10

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change
Fixes #72326

#### Describe the solution
Changes `"items":` `"chance"` field to default to 100 in 100 rather than 1 in 100 as this is very unintuitive, if you don't specify a chance you'd expect it to always spawn not almost never.
Updates the docs to reflect this
Adds `"chance": 1` to the clutter groups that appear to be the only things intentionally using the old default.
Throws an error if `"prob"` is used in `"items":` asking for it to be replaced with chance.
Attempts to replace all existing instances of this.

#### Describe alternatives you've considered
Ideally both `"item":` and `"items":` chances would work the same and ideally be x in y imo rather than 1 in x and x in 100 respectively but that would be incredibly disruptive to the codebases git history for little gain.
Short of that ideally all the now meaningless `"chance": 100`s would be removed but just in vanilla there's 1000s which again seems very disruptive for an tiny gain.

### CODE REVIEW

File: data/mods/Magiclysm/worldgen/mine_balrog.json [hunk 1]
```diff
@@ -111,22 +111,16 @@
...
+          { "item": "helmet_plate", "chance": 50, "damage": 4 },
+          { "item": "boots_plate", "chance": 75, "damage": 4 },
+          { "item": "armguard_lightplate", "chance": 60, "damage": 4 },
+          { "item": "gloves_plate", "chance": 40, "damage": 4 },
+          { "item": "legguard_lightplate", "chance": 60, "damage": 4 },
+          { "item": "armor_lc_lightplate", "chance": 70, "damage": 4 },
+          { "group": "balrog_mine_weapons", "chance": 70, "damage": 4 },
+          { "item": "bone_human", "chance": 100, "damage": 4, "count": [ 10, 20 ] }
+        ]
       }
```
Comment (Procyonae) [inline data/mods/Magiclysm/worldgen/mine_balrog.json:124]:
```suggestion
        "!":
          { "item": "balrog_mine_weapons", "chance": 70, "damage": 4 }
      },
      "item": {
        "!": [ { "item": "helmet_plate", "chance": 50, "damage": 4 },
          { "item": "boots_plate", "chance": 75, "damage": 4 },
          { "item": "armguard_lightplate", "chance": 60, "damage": 4 },
          { "item": "gloves_plate", "chance": 40, "damage": 4 },
          { "item": "legguard_lightplate", "chance": 60, "damage": 4 },
          { "item": "armor_lc_lightplate", "chance": 70, "damage": 4 },
          { "group": "balrog_mine_weapons", "chance": 70, "damage": 4 },
          { "item": "bone_human", "chance": 100, "damage": 4, "count": [ 10, 20 ] }
        ]
      }
```

File: data/json/mapgen/mansion_boarded.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
       "palettes": [ "standard_domestic_palette", "standard_domestic_landscaping_palette" ],
       "terrain": { "?": "t_water_pool_shallow_outdoors", "o": "t_window_enhanced", "$": "t_door_metal_locked" },
       "items": {
-        " ": { "item": "clutter_mansion" },
+        " ": { "item": "clutter_mansion", "chance": 1 },
```
Comment (Maleclypse) [inline data/json/mapgen/mansion_boarded.json:39]:
```suggestion
        " ": { "item": "clutter_mansion", "chance": 1 }
```

File: data/json/mapgen/mansion_boarded.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
       "palettes": [ "standard_domestic_palette", "standard_domestic_landscaping_palette" ],
       "terrain": { "?": "t_water_pool_shallow_outdoors", "o": "t_window_enhanced", "$": "t_door_metal_locked" },
       "items": {
-        " ": { "item": "clutter_mansion" },
+        " ": { "item": "clutter_mansion", "chance": 1 }
```
Comment (Maleclypse) [inline data/json/mapgen/mansion_boarded.json:39]:
```suggestion
        " ": { "item": "clutter_mansion", "chance": 1 },
```

**Comment by Procyonae** [comment]
Self reminder to do this with place_vehicles too once it's merged

**Pull request merged**